shared/fs/common/last-modified-line.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react'
+import * as Styles from '../../styles'
+import * as Kb from '../../common-adapters'
+import {formatTimeForFS} from '../../util/timestamp'
+
+export type LastModifiedLineProps = {
+  lastModifiedTimestamp?: number
+  lastWriter?: string
+  mode: 'row' | 'default' | 'menu'
+}
+
+const Username = ({mode, lastWriter}) =>
+  mode === 'row' && Styles.isMobile ? (
+    <Kb.Text type="BodySmall">{lastWriter}</Kb.Text>
+  ) : (
+    <Kb.ConnectedUsernames
+      type="BodyTinyLink"
+      usernames={[lastWriter]}
+      inline={true}
+      onUsernameClicked="profile"
+      underline={true}
+    />
+  )
+
+const LastModifiedLine = (props: LastModifiedLineProps) => {
+  const time =
+    !!props.lastModifiedTimestamp &&
+    (props.mode === 'row' ? '' : 'Last modified ') +
+      formatTimeForFS(props.lastModifiedTimestamp, props.mode !== 'row')
+  const by = !!props.lastWriter && (
+    <>
+      &nbsp;by&nbsp;
+      <Username mode={props.mode} lastWriter={props.lastWriter} />
+    </>
+  )
+  const getText = (children: React.ReactNode) => (
```

Comment:
There's so much branching here it's basically two different things, and I think this would be hard to maintain.

I think you should remove this inner closure and just do a normal:
```
if (props.mode === 'row') {
 ...
} else {
...
}
```